frontend/src/metabase/visualizations/components/Visualization/Visualization.jsx

```diff
@@ -85,34 +85,33 @@ class Visualization extends PureComponent {
     computedSettings: {},
   };
 
-  UNSAFE_componentWillMount() {
-    this.transform(this.props);
+  constructor(props) {
+    super(props);
+    this.state = this.transformPropsToState(props);
   }
 
-  UNSAFE_componentWillReceiveProps(newProps) {
+  componentDidUpdate(prevProps, prevState) {
     if (
-      !isSameSeries(newProps.rawSeries, this.props.rawSeries) ||
-      !equals(newProps.settings, this.props.settings) ||
-      !equals(newProps.timelineEvents, this.props.timelineEvents) ||
+      !isSameSeries(prevProps.rawSeries, this.props.rawSeries) ||
```

Comment:
nit: I'd swap every prop pair, so they're in the same order.

Previously, `newProps` is newer than `this.props`

Now, `this.props` is newer than `newProps`

You won't see immediate difference here, but imagine it's the logic that can't have the order swapped like `a > b`.

Reply:
fixed in https://github.com/metabase/metabase/pull/52885/commits/bf2321086f0b3237856951a76ce5f609805b7614